recipes/catch2/2.x.x/conanfile.py

```diff
@@ -35,27 +35,48 @@ def _configure_cmake(self):
         self._cmake.definitions["BUILD_TESTING"] = "OFF"
         self._cmake.definitions["CATCH_INSTALL_DOCS"] = "OFF"
         self._cmake.definitions["CATCH_INSTALL_HELPERS"] = "ON"
-        self._cmake.configure(
-            source_folder=self._source_subfolder,
-            build_folder=self._build_subfolder
-        )
+        self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
+    def build(self):
+        # Catch2 does skip install if included as subproject:
+        # https://github.com/catchorg/Catch2/blob/79a5cd795c387e2da58c13e9dcbfd9ea7a2cfb30/CMakeLists.txt#L100-L102
+        main_cml = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(main_cml, "if (NOT_SUBPROJECT)", "if (TRUE)")
+        if self.version >= "2.13.4":
+            cmake = self._configure_cmake()
+            cmake.build()
+        else:
+            no_copy_source = True
+
     def package(self):
         self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
         cmake = self._configure_cmake()
         cmake.install()
         tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
         tools.rmdir(os.path.join(self.package_folder, "share"))
         for cmake_file in ["ParseAndAddCatchTests.cmake", "Catch.cmake", "CatchAddTests.cmake"]:
-            self.copy(cmake_file,
-                      src=os.path.join(self._source_subfolder, "contrib"),
-                      dst=os.path.join("lib", "cmake", "Catch2"))
+            self.copy(
+                cmake_file,
+                src=os.path.join(self._source_subfolder, "contrib"),
+                dst=os.path.join("lib", "cmake", "Catch2"),
+            )
 
     def package_id(self):
-        self.info.header_only()
+        if self.version < "2.13.4":
+            self.info.header_only()
 
     def package_info(self):
-        self.cpp_info.builddirs = [os.path.join("lib", "cmake", "Catch2")]
+        if self.version < "2.13.4":
+            self.cpp_info.builddirs = [os.path.join("lib", "cmake", "Catch2")]
```

Comment:
the files are always copied is this required?

Reply:
This is part of the old recipe, I do not know if this is needed or not 🤷

Reply:
All I know is that I cannot do this AND the components, because CCI complains about stuff

Reply:
@prince-chrismc this line is actually required due to some cmake files being placed there by Catch2